### ISSUE TITLE
Configure issue when building in an Ubuntu 24.04 container image

### ISSUE THREAD

**Issue by theasp** [openpmix/openpmix#3389]
### What version of the PMIx Reference Library are you using? 

5.0.3

### Describe how PMIx was installed 

Source

### Please describe the system on which you are running

* Operating system/version: Ubuntu 24.04
* Computer hardware: N/A
* Network type: IB

-----------------------------

## Details of the problem

I'm building PMIX as part of a container build process for use on our Slurm cluster.  I am able to build PMIX 4.2.9 on Ubuntu 20.04 and 22.04, but 24.04 `configure` fails with the error ` cat: -: No such file or directory`.  I am able to get 5.0.3 to configure on Ubuntu 24.04 when I use `bash -x ./configure`, but not `./configure` alone or `bash -x ./configure > /tmp/some-file 2>&1`.  I've also tried running `autogen.pl` before `configure`.

If I configure PMIX interactively in a terminal it seems to work fine.  My guess is there is something unhappy with the TTY state during configure.  The containers are built using Gitlab-CI, which triggers a Slurm job, which builds the image on a node using buildah.

I'm building PMIX as part of a container build process using roughly this shell script, with `PMIX_VER=5.0.3` set in the environment for 24.04 before the build:
```bash
PMIX_URL_BASE=${PMIX_URL_BASE:-https://github.com/openpmix/openpmix/releases/download}
PMIX_VER=${PMIX_VER:-4.2.9}
PMIX_SRC_DIR="pmix-${PMIX_VER}"
PMIX_SRC_PATH="/usr/local/src/${PMIX_SRC_DIR}"
PMIX_TAR="${PMIX_SRC_DIR}.tar.bz2"
PMIX_URL="${PMIX_URL_BASE}/v${PMIX_VER}/${PMIX_TAR}"
PMIX_PREFIX=/usr/local

# ...

    cd /usr/local/src
    wget -nv "${PMIX_URL}"
    tar -xaf "${PMIX_TAR}"

    cd "${PMIX_SRC_PATH}"

    ./configure --prefix="${PMIX_PREFIX}"
    make

    cd "${PMIX_SRC_PATH}"
    make install

    cd /
    rm -rf /usr/local/src/*
```

This is the output:
```
This is the output of `./configure`
 checking for PMIX pkg-config Modules... 
 checking for PMIX pkg-config Modules.private... 
 *** Final output
 ============================================================================
 == Final compiler flags
 ============================================================================
 checking final CPPFLAGS... -iquote/usr/local/src/pmix-5.0.3 -iquote/usr/local/src/pmix-5.0.3/src -I/usr/local/src/pmix-5.0.3/include  
 checking final CFLAGS... -O3 -DNDEBUG  -finline-functions -mcx16  -fvisibility=hidden
 checking final LDFLAGS...   
 checking final LIBS... -lm -ldl -levent_core -levent_pthreads -lhwloc
 checking if libtool needs -no-undefined flag to build shared libraries... no
 ============================================================================
 == Configuration complete
 ============================================================================
 checking that generated files are newer than configure... done
 configure: creating ./config.status
 cat: -: No such file or directory
 configure: error: could not make ./config.status
```

From the output of `bash -x` it's not very clear where/why this `cat -` is happening.

**Comment by rhc54** [comment]
Check the shell where it fails and see if it is using `bash` - I suspect the default shell in Ubuntu got changed to something else. There are a few "bash-ism" in the configure script that have recently been removed, so that's the most likely culprit here.

**Comment by theasp** [comment]
There actually seems to be some degree of chance when it works.  There were a few times today (in a row!) where it worked when `./configure` was ran directly, but re-testing this later failed.  I've also not been able to get it to work with `bash -x` which worked previously, but that may have just worked once as I was relying on the layer being cached to indicate it worked.  Since `bash -x` was failing, I was able to get the spot it errors for me.

An attempt with forcing `/bin/sh` to be bash:
```
 + ln -sf /bin/bash /bin/sh
 + ls -l /bin/sh
 lrwxrwxrwx 1 root root 9 Aug 16 15:16 /bin/sh -> /bin/bash
 + ./configure --prefix=/usr/local
 checking for a sed that does not truncate output... /usr/bin/sed
 configure: builddir: /usr/local/src/pmix-5.0.3
 configure: srcdir: /usr/local/src/pmix-5.0.3
[...]
 checking that generated files are newer than configure... done
 configure: creating ./config.status
 cat: -: No such file or directory
 configure: error: could not make ./config.status
 ```
 
This is what I was able to get with `bash -x ./configure`:
 ```
 + cat
 + echo 'cat >conf784subs.awk <<_ACEOF'
 + echo 'am__EXEEXT_FALSE
 am__EXEEXT_TRUE
 LTLIBOBJS
 LIBOBJS
 libpmi2_so_version
 [...]
 PACKAGE_TARNAME
 PACKAGE_NAME
 PATH_SEPARATOR
 SHELL
 am__quote'
 + ac_delim_num=524
 + ac_delim='%!_!# '
 + for ac_last_try in false false false false false :
 + . ./conf784subs.sh
 ++ cat
 cat: -: No such file or directory
 + as_fn_error 1 'could not make ./config.status' 94990 5
 + as_status=1
 + test 1 -eq 0
 + test 5
 + as_lineno=94990
 + as_lineno_stack=as_lineno_stack=
 + printf '%s\n' 'configure:94990: error: could not make ./config.status'
 + printf '%s\n' 'configure: error: could not make ./config.status'
 configure: error: could not make ./config.status
 ```

**Comment by rhc54** [comment]
Can you try the head of either the master or v5.0 branches? There have been some changes to the configure script and perhaps one of those might impact this problem. Honestly don't know if anyone else has tried Ubunto 24.04 yet, so it may also be some new issue.

**Comment by theasp** [comment]
I can't reproduce the flex issue manually.

**Comment by theasp** [comment]
I've solved it!  The job was being submitted to slurm using `ssh -tt <host> srun ...`, after removing `-tt` it works reliably, but I added that 3 years ago with a note that indicates it's needed for passing signals.  

I cannot explain why this only affects 24.04, and only pmix (other configure scripts work fine), however I think my build environment is sufficiently "not normal" to consider this closed. 

Thanks for the help @rhc54!
